Core/Core/Planner/CalendarEvent/ViewModel/EditCustomFrequencyViewModel.swift

```diff
@@ -0,0 +1,276 @@
+            return .endDate(date)
+        case (.afterOccurrences, _, let count) where count > 0:
+            return .occurrenceCount(count)
+        default:
+            return nil
+        }
+    }
+
+    var translatedRule: RecurrenceRule? {
+        guard let end else { return nil }
+
+        var daysOfWeek: [DayOfWeek]?
+        var daysOfTheMonth: [Int]?
+        var monthsOfTheYear: [Int]?
+
+        if case .weekly = frequency {
+            daysOfWeek = daysOfTheWeek.map({ DayOfWeek($0) }).nilIfEmpty
+        }
+
+        if case .monthly = frequency, let monthDay = dayOfMonth {
+            switch monthDay {
+            case .weekday(let weekday):
+                daysOfWeek = [weekday]
+            case .day(let dayNo):
+                daysOfTheMonth = [dayNo]
+            }
+        }
+
+        if case .yearly = frequency {
+            daysOfTheMonth = dayOfYear.flatMap({ [$0.day] })
+            monthsOfTheYear = dayOfYear.flatMap({ [$0.month] })
+        }
+
+        return RecurrenceRule(
+            recurrenceWith: frequency,
+            interval: interval.value,
+            daysOfTheWeek: daysOfWeek,
+            daysOfTheMonth: daysOfTheMonth,
+            daysOfTheYear: nil,
+            weeksOfTheYear: nil,
+            monthsOfTheYear: monthsOfTheYear,
+            setPositions: nil,
+            end: end
+        )
+    }
+}
+
+// MARK: - Selected Weekdays Text
+
+extension EditCustomFrequencyViewModel {
+
+    /// ### Possible cases:
+    ///
+    /// 1. Less than 3 selected ~> returns **Plural** forms.
+    ///
+    /// Example: `[.sunday, .monday] ~> [Sundays, Mondays]`
+    ///
+    /// 2. 3 or more selected -> returns **Short** forms.
+    ///
+    /// Example: `[.sunday, .monday, .friday] ~> [Sun, Mon, Fri]`
+    ///
+    /// 3. Weekdays (Monday to Friday) selected ~> returns **"Weekdays"**
+    ///
+    /// 4. Weekdays & 1 more selected ~> returns **["Weekdays", "{DAY Short Form}"]**
+    ///
+    /// Example: `[.monday, .., .friday, .saturday] ~>  ["Weekdays", "Sat"]`
+    ///
+    /// 5. All days selected ~> returns **["Every Day of the Week"]**
+    ///
+    var selectedWeekdaysTexts: [String] {
+
+        let weekdays = daysOfTheWeek
+        var tags = [String]()
+
+        if weekdays.allDaysIncluded {
+            return [String(localized: "Every Day of the Week", bundle: .core)]
+        }
+
+        if weekdays.hasWeekdays {
+            tags.append(String(localized: "Weekdays", bundle: .core))
+
+            if let nonWeekDays = weekdays.nonWeekdays.nilIfEmpty {
+                tags.append(contentsOf: nonWeekDays.map({ $0.shortText }))
+            }
+
+        } else {
+            let fewDays = weekdays.count < 3
+            tags.append(contentsOf: weekdays.map { wday in
+                return fewDays ? wday.pluralText : wday.shortText
+            })
+        }
+
+        return tags.nilIfEmpty ?? [String(localized: "Not selected", bundle: .core)]
```

Comment:
I'm not sure, but aren't we ignoring this string because we display the PromptLabel instead of the LabelButtons?